data/modules/scripts/gamestore/init.lua

```diff
@@ -560,18 +560,41 @@ function Player.canBuyOffer(self, offer)
 				disabled = 1
 				disabledReason = "You already have an active XP boost."
 			end
+		elseif offer.type == GameStore.OfferTypes.OFFER_TYPE_HIRELING then
+			if self:getHirelingsCount() == 10 then
```

Comment:
```suggestion
			if self:getHirelingsCount() >= 10 then
```
Even if it isn't expected to happen, isn't better to check if it's equal or higher than 10?
So even if somehow player manages to get an hypothetical eleventh Hireling the code will keep working.